### ISSUE TITLE
[JENKINS-2909] Endless authentication to SVN when invalid user/password

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/subversion-plugin#520]
<p>I found this in hudson 235 and it is still in hudson 277.</p>

<p>How to recreate it:</p>
<ul class="alternate" type="square">
	<li>Create project with SVN Repository (http protocol and user/name<br/>
authentication)</li>
	<li>Tick Poll SCM for emaple to '* * * * *'</li>
	<li>After a few minutes change the password of the SVN user you used in<br/>
authentication (Do it in your SVN environment, NOT in hudson - hudson will not <br/>
know anything about password 'expiration')<br/>
Observed results:<br/>
Polling will start endless authentication tries... (about 5-10 requests per <br/>
sec.)<br/>
Expected results:<br/>
After one or two attempts poll should stop trying to authenticate</li>
</ul>

<p>The problem is in method<br/>
SubversionSCM.SVNAuthenticationProviderImpl.requestClientAuthentication(...) {<br/>
}<br/>
It always returns credentials (invalid), however, it should return it only once <br/>
(at the first attempt to authentication). Other words should be aware of input <br/>
parameters: SVNErrorMessage errorMessage, SVNAuthentication previousAuth</p>

<p>Witold Delekta</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=witolddelekta">witolddelekta</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-2909" target="_blank">Endless authentication to SVN when invalid user/password</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: subversion-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2009-06-24T10:04:41+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 20251212-090250
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I found this in hudson 235 and it is still in hudson 277.</p>

<p>How to recreate it:</p>
<ul class="alternate" type="square">
	<li>Create project with SVN Repository (http protocol and user/name
authentication)</li>
	<li>Tick Poll SCM for emaple to '* * * * *'</li>
	<li>After a few minutes change the password of the SVN user you used in
authentication (Do it in your SVN environment, NOT in hudson - hudson will not 
know anything about password 'expiration')
Observed results:
Polling will start endless authentication tries... (about 5-10 requests per 
sec.)
Expected results:
After one or two attempts poll should stop trying to authenticate</li>
</ul>

<p>The problem is in method
SubversionSCM.SVNAuthenticationProviderImpl.requestClientAuthentication(...) {
}
It always returns credentials (invalid), however, it should return it only once 
(at the first attempt to authentication). Other words should be aware of input 
parameters: SVNErrorMessage errorMessage, SVNAuthentication previousAuth</p>

<p>Witold Delekta</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Platform: All, OS: All</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>dodok1:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-2909?focusedId=128387&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128387">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 3885 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

</pre>
</details>

</ul>
</details>

<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 3885 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-2909?focusedId=128390&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128390">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in hudson
User: : kohsuke
Path:
 trunk/www/changelog.html
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19206" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19206</a>
Log:
recording <a href="https://issues.jenkins.io/browse/JENKINS-2909" title="Endless authentication to SVN when invalid user/password" class="issue-link" data-issue-key="JENKINS-2909"><del>JENKINS-2909</del></a> for 1.313</p>
</pre>
</details>

</ul>
</details>

<p>Code changed in hudson<br/>
User: : kohsuke<br/>
Path:<br/>
 trunk/www/changelog.html<br/>
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19206" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19206</a><br/>
Log:<br/>
recording <a href="https://issue-redirect.jenkins.io/issue/2909" title="Endless authentication to SVN when invalid user/password" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/2909">JENKINS-2909</a>"><del>JENKINS-2909</del></a> for 1.313</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-2909?focusedId=128391&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128391">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in hudson
User: : kohsuke
Path:
 trunk/hudson/main/war/pom.xml
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19207" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19207</a>
Log:
recording <a href="https://issues.jenkins.io/browse/JENKINS-2909" title="Endless authentication to SVN when invalid user/password" class="issue-link" data-issue-key="JENKINS-2909"><del>JENKINS-2909</del></a> for 1.313</p>
</pre>
</details>

</ul>
</details>

<p>Code changed in hudson<br/>
User: : kohsuke<br/>
Path:<br/>
 trunk/hudson/main/war/pom.xml<br/>
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19207" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19207</a><br/>
Log:<br/>
recording <a href="https://issue-redirect.jenkins.io/issue/2909" title="Endless authentication to SVN when invalid user/password" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/2909">JENKINS-2909</a>"><del>JENKINS-2909</del></a> for 1.313</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `is_duplicated_by` from Jira: <a class="jira-relationship" href="https://issue-redirect.jenkins.io/issue/3885">JENKINS-3885</a>]</i>
